### PR TITLE
[OSCI][BUG] Fixing bug for making playground toggle button available in OUI v2.0

### PR DESCRIPTION
**Pull request by BigSamu**
…round toggle button visible

### Description
 Changing the definition for const 'isPlaygroundUnsupported' to make the playground toggle button visible in OUI 2.0. This bug didn't appear until a change was brought in the commit [02d0b4, line 136](https://github.com/BigSamu/oui/commit/0d20b45fc7735d8de5cb2fc65099089d35ffdef2#diff-fc77ffcd658b5a0d6c328efdb465259b23c110ce6cc726e338690218d540b937R136). Essentially, before that change, the variable `isPlaygroundUnsupported` was always true if you were using a browser different from Explorer.

### Issues Resolved
Issue #1140

### Check List
- [x] New functionality includes testing.
- [ ] New functionality has been documented.
- [x] All tests pass
  - [x] `yarn lint`
  - [x] `yarn test-unit`
- [x] Update [CHANGELOG.md](./../CHANGELOG.md)
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/oui/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -24,6 +24,7 @@
 - Extract build archive into a folder for OSD integration test CI ([#1075](https://github.com/opensearch-project/oui/pull/1075))
 - Correct file path for import of Query component ([#1069](https://github.com/opensearch-project/oui/pull/1069))
 - Fix "Guidelines" documentation links rendering blank pages ([#1111](https://github.com/opensearch-project/oui/pull/1111))
+- Fix bug for making playground toggle button available in OUI v2.0 ([#1162](https://github.com/opensearch-project/oui/pull/1162))
```
Comment (BSFishy) [inline CHANGELOG.md:27]:
```suggestion
- Fix playground support check ([#1162](https://github.com/opensearch-project/oui/pull/1162))
```

File: src-docs/src/components/guide_section/guide_section.tsx [hunk 2]
```diff
@@ -133,7 +133,7 @@ export const GuideSection: FunctionComponent<GuideSection> = ({
 
   const renderPlaygroundToggle = () => {
     const isPlaygroundUnsupported =
-      typeof window !== 'undefined' && typeof document !== 'undefined';
+      typeof window === 'undefined' || typeof document === 'undefined';
 
     if (!isPlaygroundUnsupported && !!playground) {
```
Comment (BSFishy) [inline src-docs/src/components/guide_section/guide_section.tsx:138]: I feel like a `isPlaygroundSupported` would be clearer than `isPlaygroundUnsupported`, but I'm not going to block over something that trivial

**Review by BSFishy — APPROVED** [review]

File: src-docs/src/components/guide_section/guide_section.tsx (hunk 2 above)
Comment (BigSamu) [reply src-docs/src/components/guide_section/guide_section.tsx:138]: Agree! Let's see what Josh says. Indeed, that minor issue could bring missread when defining that variable.

**Comment by BigSamu** [comment]
@joshuarrrr, @BSFishy  

PR updated with `main` branch. Is anything else required to approve this Pull Request?

**Comment by BSFishy** [comment]
> PR updated with `main` branch. Is anything else required to approve this Pull Request?

LGTM, just need another reviewer

**Review by ashwin-pc — APPROVED** [review]

**Pull request merged**